### PR TITLE
RNS Shell Implementation

### PR DESCRIPTION
**Pull request by munezbn**
Implemented RNS Shell for replacing sk_app which was used from Skia for POC purpose.
Supports Only Linux Platform now.
Has Boot strap code for application.
Has platform Task runner implementation.
Has Layer management and Render Tree management.
Has Platform Graphics support with OpenGL backend with X11 & LibWPE client backend. Supports GLX and EGL interface.
Has compositor framework for rendering Render Layer tree.

Added RNS logging and Utility Helper Macros.

### CODE REVIEW

File: ReactSkia/BUILD.gn [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
+  # Enable API profiling
+  rns_enable_api_profiling = false
+}
+
 config("ReactSkia_config") {
   include_dirs = [
-    # "//react-native",
+     "//rns_shell/common",
+     "//rns_shell/platform/graphics",
+     "//rns_shell/platform/graphics/gl",
```
Comment (Kudo) [inline ReactSkia/BUILD.gn]: could use `public_configs` in rns_shell. something like the
https://github.com/Kudo/react-native-skia/blob/f0b100c12acd0c934ba32f213601df90c8c685b9/build/secondary/react-native/ReactCommon/yoga/BUILD.gn#L34

**Review by Kudo — CHANGES_REQUESTED** [review]
if sk_app is not used, please try to remove sk_app, so that git may able to know the file renaming. otherwise, it's hard to review the difference between sk_app and rns_shell.

File: ReactSkia/BUILD.gn (hunk 1 above)
Comment (munezbn) [reply ReactSkia/BUILD.gn]: Done.

**Comment by munezbn** [comment]
> if sk_app is not used, please try to remove sk_app, so that git may able to know the file renaming. otherwise, it's hard to review the difference between sk_app and rns_shell.

Done, But I noticed that git didn't detect all the files as renamed, some of file still shown as new files. But it is better than before.

**Review by Kudo — APPROVED** [review]
LGTM 🚀

**Pull request merged**